usaspending_api/references/v2/views/cfda.py

```diff
@@ -1,32 +1,33 @@
-from rest_framework.response import Response
 from rest_framework.views import APIView
 from requests import post
-from time import sleep
+from rest_framework.response import Response
 from django.conf import settings
 from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.exceptions import NoDataFoundException
+import logging
 
-CFDA_DICTIONARY = None
+logger = logging.getLogger(__name__)
 
 
 class CFDAViewSet(APIView):
     """
-    Return an agency name and active fy.
+    Returns opportunity totals for a CFDA or all opportunity totals.
     """
 
     endpoint_doc = "usaspending_api/api_contracts/contracts/v2/references/cfda/totals.md"
 
     @cache_response()
     def get(self, request, cfda=None):
         """
-        Return the view's queryset.
+        Return the CFDA's opportunity totals from grants.gov, or totals for all current CFDAs.
         """
-        self._populate_cfdas_if_needed()
+
+        CFDA_DICTIONARY = self.get_cfdas()
```

Comment:
It appears this code will not longer "cache" the CFDA data as it did before using the global variable which persists as long and the uWSGI worker is running the django app